lib/routes/routesUtils.js

```diff
@@ -428,19 +428,30 @@ const routesUtils = {
      * @param {string} [routingInfo.prefixFromRule] - key prefix to be replaced
      * @param {string} objectKey - key name (may have been modified in
      * websiteGet api to include index document)
+     * @param {string} indexDocument - index document's name
      * @param {boolean} encrypted - whether request was https
      * @param {object} response - response object
+     * @param {string} hostHeader - host's header from headers.host
      * @param {object} log - Werelogs instance
      * @return {undefined}
      */
-    redirectRequest(routingInfo, objectKey, encrypted, response, log) {
+    redirectRequest(routingInfo, objectKey, indexDocument, encrypted, response,
+      hostHeader, log) {
         const { hostName, protocol, httpRedirectCode, replaceKeyPrefixWith,
             replaceKeyWith, prefixFromRule } = routingInfo;
 
         const redirectProtocol = protocol || encrypted ? 'https' : 'http';
         const redirectCode = httpRedirectCode || 302;
+        const redirectHostName = hostName || hostHeader;
 
-        let redirectKey = objectKey === undefined ? '' : objectKey;
+        let redirectKey;
+        if (objectKey === undefined) {
+            redirectKey = '';
+        } else if (indexDocument) {
+            redirectKey = objectKey.replace(indexDocument, '');
```

Comment:
Does replacing the index key with an empty string fix some tests? Which tests were not passing with the previous behavior?